app/src/main/kotlin/com/flxrs/dankchat/preferences/notifications/highlights/HighlightsScreen.kt

```diff
@@ -461,6 +493,134 @@ private fun UserHighlightItem(
     }
 }
 
+@Composable
+private fun BadgeHighlightItem(
+    item: BadgeHighlightItem,
+    onChanged: (BadgeHighlightItem) -> Unit,
+    onRemove: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    ElevatedCard(modifier) {
+        Row {
+            Column(
+                modifier = Modifier
+                    .weight(1f)
+                    .padding(8.dp),
+            ) {
+                if (item.isCustom) {
+                    OutlinedTextField(
+                        modifier = Modifier.fillMaxWidth(),
+                        value = item.badgeName,
+                        onValueChange = { onChanged(item.copy(badgeName = it)) },
+                        label = { Text(stringResource(R.string.badge)) },
+                        keyboardOptions = KeyboardOptions(imeAction = ImeAction.Done),
+                        maxLines = 1,
+                    )
+                } else {
+                    var name = ""
+                    when (item.badgeName) {
+                        "broadcaster"-> name = stringResource(R.string.badge_broadcaster)
+                        "admin"-> name = stringResource(R.string.badge_admin)
+                        "staff"-> name = stringResource(R.string.badge_staff)
+                        "moderator"-> name = stringResource(R.string.badge_moderator)
+                        "partner"-> name = stringResource(R.string.badge_verified)
+                        "vip"-> name = stringResource(R.string.badge_vip)
+                        "founder"-> name = stringResource(R.string.badge_founder)
+                        "subscriber"-> name = stringResource(R.string.badge_subscriber)
+                    }
+                    OutlinedTextField(
```

Comment:
We should use a `Text` here, since a "readOnly" text field unfortunately is still focusable which is kind of confusing UX, imo.